atomic_reactor/plugin.py

```diff
@@ -210,6 +214,9 @@ def __init__(self, dt, workflow, plugin_class_name, plugins_conf, *args, **kwarg
         self.workflow = workflow
         super(BuildPluginsRunner, self).__init__(plugin_class_name, plugins_conf, *args, **kwargs)
 
+    def on_plugin_failed(self):
+        self.workflow.build_has_failed = True
```

Comment:
This is about whether atomic-reactor succeeded or failed, not just whether the docker build did.

Some plugins can cause the entire run to be a failure, and there needs to be a hook to allow different behaviour in that case.